substrate/frame/support/src/generate_genesis_config.rs

```diff
@@ -208,89 +217,154 @@ pub fn retain_initialized_fields(
 #[macro_export]
 macro_rules! build_struct_json_patch {
 	(
-		$($struct_type:ident)::+ { $($tail:tt)* }
+		$($struct_type:ident)::+ { $($body:tt)* }
 	) => {
 		{
-			let mut keys = $crate::__private::Vec::<$crate::generate_genesis_config::InitializedField>::default();
+			let mut __keys = $crate::__private::Vec::<$crate::generate_genesis_config::InitializedField>::default();
```

Comment:
to not conflicts with ident used by the caller. Like if the GenesisConfig has a `keys` field or some `keys` variable in scope.

Reply:
ah, noice. Thanks!

Reply:
Yes, this is exactly to avoid conflicts with the user code. But I did quick research and did not find any unspoken rules that variable declared by macro should be underscored.

In this macro only some of them could actually conflict with the user code identifiers, so I decided to underscore them. Then to keep the naming consistent across entire macro I renamed all the variables to use the same naming scheme. Hope this makes sense.